x-pack/test/functional/services/machine_learning/job_wizard_common.ts

```diff
@@ -358,6 +388,39 @@ export function MachineLearningJobWizardCommonProvider({ getService }: FtrProvid
       await this.assertDateRangeSelection(expectedStartDate, expectedEndDate);
     },
 
+    async ensureAdditionalSettingsSectionOpen() {
+      await retry.tryForTime(5000, async () => {
+        if ((await testSubjects.exists(additionalSettingsSectionSelector())) === false) {
+          await testSubjects.click('mlJobWizardToggleAdditionalSettingsSection');
+          await testSubjects.existOrFail(additionalSettingsSectionSelector(), { timeout: 1000 });
+        }
+      });
+    },
+
+    async ensureNewCustomUrlFormModalOpen() {
+      await retry.tryForTime(5000, async () => {
+        if ((await testSubjects.exists(newCustomUrlFormModal())) === false) {
+          await testSubjects.click('mlJobOpenCustomUrlFormButton');
+          await testSubjects.existOrFail(newCustomUrlFormModal(), { timeout: 1000 });
+        }
+      });
+    },
+
+    async addCustomUrl() {
+      await this.ensureNewCustomUrlFormModalOpen();
```

Comment:
This should call `ensureAdditionalSettingsSectionOpen` first just to be less dependent on the state of the UI.

Reply:
Added in https://github.com/elastic/kibana/pull/52269/commits/94da14ea402f329134c8cb021231ca588b7f5890